stack/rest/src/main/java/org/apache/usergrid/rest/management/ManagementResource.java

```diff
@@ -441,6 +464,229 @@ public Viewable handleAuthorizeForm( @Context UriInfo ui, @FormParam( "response_
+                        applicationCreator.createSampleFor( ownerOrgInfo.getOrganization() );
+
+                        userId = ownerOrgInfo.getOwner().getUuid();
+                        userInfo = ownerOrgInfo.getOwner();
+
+                    } else {
+
+                        // already created user, so just create an org
+                        final OrganizationInfo organization = management.createOrganization( orgName, userInfo, true );
+
+                        management.activateOrganization( organization ); // redundant?
+                        applicationCreator.createSampleFor( organization );
+                    }
+                }
+
+            }
+
+            // store the external access_token as if it were one of our own
+            management.importTokenForAdminUser( userId, extAccessToken, ttl );
+
+            // success! return JSON object with access_token field
+            accessInfo = new AccessInfo()
+                    .withExpiresIn( tokens.getMaxTokenAgeInSeconds( extAccessToken ) )
+                    .withAccessToken( extAccessToken );
+
+        } catch (Exception e) {
+            logger.debug("Error validating external token", e);
+            throw e;
+        }
+
+        return Response.status( SC_OK ).type( jsonMediaType( callback ) ).entity( accessInfo ).build();
+    }
+
+
+    /**
+     * Look up Admin User via UG Central's /management/me endpoint.
+     *
+     * @param extAccessToken Access token issued by UG Central of Admin User
+     * @return JsonNode representation of AccessInfo object for Admin User
+     * @throws EntityNotFoundException if access_token is not valid.
+     */
+    private JsonNode getMeFromUgCentral( String extAccessToken )  throws EntityNotFoundException {
+
+        // create URL of central Usergrid's /management/me endpoint
+
+        String externalUrl = properties.getProperty( USERGRID_CENTRAL_URL ).trim();
+
+        // be lenient about trailing slash
+        externalUrl = !externalUrl.endsWith( "/" ) ? externalUrl + "/" : externalUrl;
+        String me = externalUrl + "management/me?access_token=" + extAccessToken;
+
+        // use our favorite HTTP client to GET /management/me
+
+        ClientConfig clientConfig = new DefaultClientConfig();
+        clientConfig.getFeatures().put( JSONConfiguration.FEATURE_POJO_MAPPING, Boolean.TRUE);
+        Client client = Client.create(clientConfig);
+        final JsonNode accessInfoNode;
+        try {
+            accessInfoNode = client.resource( me )
+                    .type( MediaType.APPLICATION_JSON_TYPE)
+                    .get(JsonNode.class);
+
+        } catch ( Exception e ) {
+            // user not found 404
+            String msg = "Cannot find Admin User associated with " + extAccessToken;
+            throw new EntityNotFoundException( msg, e );
+        }
+        return accessInfoNode;
+    }
+
+
+    /**
```

Comment:
I moved the "superuser-only login when external token validation enabled" logic in the below new method: